pubsub.go

```diff
@@ -61,19 +70,44 @@ func KeyToTopic(key string) string {
 // The constructor interface is complicated by the need to bootstrap the pubsub topic.
 // This could be greatly simplified if the pubsub implementation handled bootstrap itself
 func NewPubsubValueStore(ctx context.Context, host host.Host, cr routing.ContentRouting, ps *pubsub.PubSub, validator record.Validator) *PubsubValueStore {
-	return &PubsubValueStore{
+	psValueStore := &PubsubValueStore{
 		ctx: ctx,
 
-		ds:   dssync.MutexWrap(ds.NewMapDatastore()),
-		host: host, // needed for pubsub bootstrap
-		cr:   cr,   // needed for pubsub bootstrap
-		ps:   ps,
+		ds:                      dssync.MutexWrap(ds.NewMapDatastore()),
+		ps:                      ps,
+		host:                    host,
+		rebroadcastInitialDelay: 100 * time.Millisecond,
+		rebroadcastInterval:     time.Second,
 
 		subs:     make(map[string]*pubsub.Subscription),
 		watching: make(map[string]*watchGroup),
 
 		Validator: validator,
 	}
+
+	host.SetStreamHandler(OnJoinProto, func(s net.Stream) {
+		defer net.FullClose(s)
+
+		msgData, err := readBytes(s)
+		if err != nil {
+			return
+		}
+
+		msg := pb.RequestLatest{}
+		if msg.Unmarshal(msgData) != nil {
+			return
+		}
+
+		response, err := psValueStore.getLocal(*msg.Identifier)
+		if err != nil {
+			return
+		}
+
+		if writeBytes(s, response) != nil {
```

Comment:
If we keep the existing structure and don't change the protocol to be send-only instead of send-response (https://github.com/libp2p/go-libp2p-pubsub-router/pull/33/files#r300765803), do we want any sort of protobuf or encapsulation on the response data or is raw bytes here acceptable?